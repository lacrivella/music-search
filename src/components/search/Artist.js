import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Artist({ artist }) {
  return (
    <Link to={`/artist/${artist.id}`}>
    <section>
      <p>{artist.name}</p>
      <p>{artist.disambiguation}</p>
    </section>
    </Link>
  );
}

Artist.propTypes = {
  artist: PropTypes.shape({
    name: PropTypes.string.isRequired,
    disambiguation: PropTypes.string.isRequired,
    id: PropTypes.string
  })
};

export default Artist;
