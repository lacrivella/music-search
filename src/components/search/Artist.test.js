import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artist component', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist artist={{ name: 'WASP', disambiguation: 'Metal Band', id: 'whatev' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});