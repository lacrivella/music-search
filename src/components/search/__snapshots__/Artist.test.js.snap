// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders Artist 1`] = `
<Link
  to="/artist/WASP/whatev"
>
  <section>
    <p>
      WASP
    </p>
    <p>
      Metal Band
    </p>
  </section>
</Link>
`;
